File: url-ping-go/handler.go
package function

import (
	"io/ioutil"
	"net/http"
)

// Handle a serverless request
func Handle(req []byte) string {
	if len(req) == 0 {
		return "Pass a URL for ping"
	}

	res, err := http.Get(string(req))
	if err != nil {
		return err.Error()
	}

	bytesOut, _ := ioutil.ReadAll(res.Body)

	return string(bytesOut)
}

File: url-ping-js/handler.js
"use strict"

const request = require("request");

module.exports = (context, callback) => {

    
    if(!context||!context.length) {
        callback(undefined, {status: "submit a URL for the ping"});    
    }

    request.get(context, (err, res, body) => {
        if(err) {
            return callback(err, {});    
        }

        return callback(undefined, {status: "done", code: res.statusCode });
    });
}
